kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -150,7 +150,16 @@ private static List<FilterDefinition> filterDefinitions(Clusters cluster, Contex
                     .orElseThrow(() -> unknownFilterKind(cluster, filterRef));
 
             var filt = filterResourceFromRef(cluster, context, filterRef);
-            return filterDefFromFilterResource(matchingFilterApi.className(), filt);
+            if ("filter.kroxylicious.io".equals(matchingFilterApi.group())
+                    && "Filter".equals(matchingFilterApi.kind())) {
+                Map<String, Object> spec = (Map<String, Object>) filt.getAdditionalProperties().get("spec");
+                String type = (String) spec.get("type");
+                Object config = spec.get("config");
+                return new FilterDefinition(type, config);
+            }
+            else {
+                return filterDefFromFilterResource(matchingFilterApi.className(), filt);
```

Comment:
maybe we should make this throw a not implemented exception initially, and reimplement if we add back in support for other CRs.

Reply:
I think this code is actually unreachable, because the code on line 150 will throw UnknownFilterKind in this case (because when we create the RuntimeDecl in io.kroxylicious.kubernetes.operator.OperatorMain#main we only configure it with the `Filter` kind). We could remove that too, if we don't think we'll be able to add support for CRD-per-Filter any time soon. But TBH I'm inclined to be optimistic for now and keep this code in there for the time being.

Reply:
Shaping the code to support multiple Filter kinds in future is fine with me, using the RuntimeDecl class and a list of kinds. The `ProxyReconciler` is also prepared to handle multiple kinds.

But I think this code is the only place where we specialize the behaviour to handle other kinds (where we can't extract the classname). This is the bit I would remove for now (and maybe the className of FilterApiDecl). It should be easy to restore the funtionality if we come back to it, along with re-introducing tests covering this code.